lib/api/core/models/repositories/repository.js

```diff
@@ -1,344 +1,421 @@
-  var
-    options = opts || {},
-    key = options.key || this.cacheKeyPrefix + object._id,
-    ttl = this.ttl;
-
-  if (options.ttl !== undefined) {
-    ttl = options.ttl;
+    key = options.key || this.getCacheKey(id, collection);
+
+    return this.cacheEngine.remove(key);
   }
 
-  if (ttl === false) {
-    return this.cacheEngine.persist(key);
+  /**
+   * @param {object} object
+   * @param {object} [options] - optional options for the current operation
+   * @param {string} [collection]
+   * @returns {*}
+   */
+  refreshCacheTTL (object, options = {}, collection = this.collection) {
+    const ttl = options.ttl !== undefined ? options.ttl : this.ttl;
+    let key;
+
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
+    }
+
+    key = options.key || this.getCacheKey(object._id, collection);
+
+
+    if (ttl === false) {
+      return this.cacheEngine.persist(key);
+    }
+
+    return this.cacheEngine.expire(key, ttl);
   }
 
-  return this.cacheEngine.expire(key, ttl);
-};
+  /**
+   * @param {object} object
+   * @param {object} [options] - optional options for the current operation
+   * @param {string} [collection]
+   * @returns {*}
+   */
+  expireFromCache (object, options = {}, collection = this.collection) {
+    const key = options.key || this.getCacheKey(object._id, collection);
 
-Repository.prototype.expireFromCache = function repositoryExpireFromCache (object, opts) {
-  var
-    options = opts || {},
-    key = options.key || this.cacheKeyPrefix + object._id;
+    return this.cacheEngine.expire(key, -1);
+  }
 
-  return this.cacheEngine.expire(key, -1);
-};
+  /**
+   * Serializes the object before being persisted to cache.
+   *
+   * @param {object} object - The object to serialize
+   * @returns {object}
+   */
+  serializeToCache (object) {
+    return _.assign({}, object);
+  }
 
-/**
- * Serializes the object before being persisted to cache.
- *
- * @param {object} object The object to serialize
- * @returns {object}
- */
-Repository.prototype.serializeToCache = function repositorySerializeToCache (object) {
-  return _.assign({}, object);
-};
+  /**
+   * Serializes the object before being persisted to the database.
+   *
+   * @param {object} object - The object to serialize
+   * @returns {object}
+   */
+  serializeToDatabase (object) {
+    return object;
+  }
 
-/**
- * Serializes the object before being persisted to the database.
- *
- * @param {object} object The object to serialize
- * @returns {object}
- */
-Repository.prototype.serializeToDatabase = function repositorySerializeToDatabase (object) {
-  return object;
-};
+  /**
+   * @param {string} id
+   * @param {string} [collection]
+   */
+  getCacheKey (id, collection = this.collection) {
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
```

Comment:
It can not happen due to a bad request, it might come from an bad plugin implementation, will use this one instead.